### PR TITLE
flow: use ebpf for socket info

### PR DESCRIPTION
**Pull request by lebauce**

### CODE REVIEW

**Comment by lebauce** [comment]
run skydive-unit-tests

**Comment by lebauce** [comment]
run skydive-scale-tests

**Comment by lebauce** [comment]
run skydive-functional-tests-backend-orientdb

**Comment by lebauce** [comment]
run tests

**Comment by lebauce** [comment]
run tests

**Comment by lebauce** [comment]
run tests

**Comment by lebauce** [comment]
ping @safchain @nplanel

File: flow/enhancers/socket_info.go [hunk 1]
```diff
@@ -28,368 +30,310 @@ import (
...
+	return fmt.Sprintf("%s:%04X", ipToHex(ip), port)
 }
 
-var rProcPidTid = regexp.MustCompile("/proc/(\\d+)/task/(\\d+)$")
-
-func parseProcStat(procProcessRoot string, si *SocketInfo) {
-	// Retreive binary name
-	exe, err := os.Readlink(procProcessRoot + "/exe")
+func getProcessInfo(pid int) (*flow.SocketInfo, error) {
+	processPath := fmt.Sprintf("/proc/%d", pid)
```
Comment (nplanel) [inline flow/enhancers/socket_info.go]: we losing the threadID (tid) here, a task could have a different name than the pid
would be possible to retrieve the info from tcptrace

File: flow/enhancers/socket_info.go [hunk 2]
```diff
@@ -28,368 +30,310 @@ import (
...
+		return err
 	}
-	return &flow.SocketInfo{
-		Process: si.Exe,
-		Pid:     int64(si.Pid),
-		Ppid:    int64(si.PPid),
-		Name:    si.Name,
-		Tid:     int64(si.Tid),
+
+	for _, item := range d {
```
Comment (nplanel) [inline flow/enhancers/socket_info.go]: you don't need to parse all /proc/..../net ; check the inode and scan only once

File: flow/enhancers/socket_info.go [hunk 3]
```diff
@@ -28,368 +30,310 @@ import (
...
+	s.remote.Flush()
+
+	inodePids := make(map[int]int)
+	namespaces := make(map[uint64]bool)
+	processes := make(map[int]*flow.SocketInfo)
+
+	buildInodePidMap := func() error {
+		// Loop through all fd dirs of process on /proc to compare the inode and
+		// get the pid. Shamelessly taken from github.com/drael/GOnetstat
+		d, err := filepath.Glob("/proc/[0-9]*/fd/[0-9]*")
```
Comment (nplanel) [inline flow/enhancers/socket_info.go]: I think Glob() is quite slow (could be check with strace) as it do lot's of stat on each file, compare to the os.Open() readdirnames() ...

**Pull request merged**